### ISSUE TITLE
bad for  react17+

### ISSUE THREAD

**Issue by liwei8** [frontend-collective/react-sortable-tree#855]
# Reporting a Bug?

Please include either a failing unit test or a simple reproduction. You can start by forking the [CodeSandbox example](https://codesandbox.io/s/wkxvy3z15w)

There was an error  ， when react 17+  react-dom17+

**Comment by mmvcode** [comment]
Yep, just started getting the following error after upgrading to React 17:

`Unable to find node on an unmounted component.`

Seems like it can't find `.root`?

https://codesandbox.io/s/upbeat-lovelace-bstd2?file=/src/App.js

**Comment by crazyhtm** [comment]
Same for me:

├── UNMET PEER DEPENDENCY react@17.0.1 
└─┬ react-sortable-tree@2.8.0
  └─┬ frontend-collective-react-dnd-scrollzone@1.0.2
    └── react@16.14.0 

npm ERR! peer dep missing: react@^16.3.0, required by react-sortable-tree@2.8.0

Any workaround?

**Comment by umberto-sonnino** [comment]
+1!
Can this be updated to be compatible with React 17?

**Comment by nhat-tr** [comment]
#821 you can check this thread, there is a workaround

**Comment by Flibo** [comment]
For what it's worth, older versions of this package seem to cooperate better with React 17. I'm using `"react-sortable-tree": "^1.8.1"` and it appears to be working.

**Comment by arcataroger** [comment]
The workarounds in #821 didn't work for me.

Not sure if this package is abandoned, but there's a fork on NPM that's updated for React 17:
https://www.npmjs.com/package/react-sortable-tree-patch-react-17

**Comment by nosferatu500** [comment]
Hi guys,

I fixed the issues with React 17 and created a separate package.

You can install it using
```
yarn add @nosferatu500/react-sortable-tree
```

But it does not include types for typescript.
you can workaround with it:

- create `decs.d.ts` file and place it with `tsconfig.json`
- add `declare module @nosferatu500/react-sortable-tree` to `decs.d.ts` file
- in `tsconfig.json` change
```
"include": [
  "src"
  ]
```

to

```
"include": [
  "src",
  "decs.d.ts"
  ]
```

I am not planning any new features or any other updates (only bug fixes and only if it block my own work), but if you want to improve something, you can send me the PR to me https://github.com/nosferatu500/react-sortable-tree and I will merge it.
